### PR TITLE
Add missing TR::CodeGenerator() constructor

### PR DESCRIPTION
**Pull request by DanHeidinga**
Fix `snapshot` build break.

Signed-off-by: Dan Heidinga <heidinga@redhat.com>

### CODE REVIEW

**Comment by DanHeidinga** [comment]
travis failure is the ddr build.

**Comment by DanHeidinga** [comment]
@0xdaryl are you able to review this?

**Review by fjeremic — APPROVED** [review]

File: runtime/compiler/codegen/CodeGenerator.hpp [hunk 1]
```diff
@@ -35,6 +35,10 @@ class OMR_EXTENSIBLE CodeGenerator : public J9::CodeGeneratorConnector
 
    CodeGenerator(TR::Compilation *comp) :
       J9::CodeGeneratorConnector(comp) {}
+
+   CodeGenerator() :
+      J9::CodeGeneratorConnector(TR::comp()) {}
+
```
Comment (fjeremic) [inline runtime/compiler/codegen/CodeGenerator.hpp:41]: Actually I'll delegate to @0xdaryl on this one. I don't see this constructor in the master branch. Not sure what it is for, but `TR::comp()` is rather expensive so I think Daryl will have better gauge of this change.

Comment (DanHeidinga) [reply runtime/compiler/codegen/CodeGenerator.hpp:41]: It was introduced in the `snapshot` branch of the openj9-omr repo:
https://github.com/eclipse/openj9-omr/blob/852aba7251a37b785df5282768d2e8907ba4eda7/compiler/codegen/CodeGenerator.hpp#L43-L44

Comment (fjeremic) [reply runtime/compiler/codegen/CodeGenerator.hpp:41]: Thanks for pointing that. Agreed, change is fine!

**Pull request merged**